core/src/klio_core/options.py

```diff
@@ -0,0 +1,265 @@
+    if args:
+        return wrapper(args[0])
+    return wrapper
+
+
+#####
+# common options
+#####
+def direct_runner(func):
+    return click.option(
+        "--direct-runner",
+        default=False,
+        is_flag=True,
+        help="Run the job locally via the DirectRunner.",
+    )(func)
+
+
+def update(func):
+    return click.option(
+        "--update/--no-update",
+        default=None,
+        is_flag=True,
+        help="[Experimental] Update an existing streaming Cloud Dataflow job.",
+    )(func)
+
+
+def show_logs(func):
+    return click.option(
+        "--show-logs",
+        default=False,
+        show_default=True,
+        is_flag=True,
+        help="Show a job's logs while profiling.",
+    )(func)
+
+
+#####
+# options for `klioexec profile memory
+#####
+def interval(func):
+    return click.option(
+        "--interval",
+        default=0.1,
+        show_default=True,
+        type=float,
+        help="Sampling period (in seconds).",
+    )(func)
+
+
+def include_children(func):
+    return click.option(
+        "--include-children",
+        default=False,
+        show_default=True,
+        is_flag=True,
+        help="Monitor forked processes as well (sums up all process memory).",
+    )(func)
+
+
+def multiprocess(func):
+    return click.option(
+        "--multiprocess",
+        default=False,
+        show_default=True,
+        is_flag=True,
+        help=(
+            "Monitor forked processes creating individual plots for each "
+            "child."
+        ),
+    )(func)
+
+
+def plot_graph(func):
+    return click.option(
+        "-g",
+        "--plot-graph",
+        default=False,
+        show_default=True,
+        is_flag=True,
+        help=(
+            "Plot memory profile using matplotlib. Saves to "
+            "klio_profile_memory_<YYYYMMDDhhmmss>.png."
+        ),
+    )(func)
+
+
+#####
+# options for `klio job profile memory-per-line`
+#####
+def maximum(func):
+    return click.option(
+        "--maximum",
+        "get_maximum",
+        default=False,
+        show_default=True,
+        is_flag=True,
+        cls=MutuallyExclusiveOption,
+        mutually_exclusive=["per_element"],
+        help=(
+            "Print maximum memory usage per line in aggregate of all input "
+            "elements process."
+        ),
+    )(func)
+
+
+def per_element(func):
+    return click.option(
+        "--per-element",
+        default=False,
+        show_default=True,
+        is_flag=True,
+        cls=MutuallyExclusiveOption,
+        mutually_exclusive=["maximum"],
+        help="Print memory usage per line for each input element processed.",
+    )(func)
+
+
+#####
+# options for `klioexec profile timeit`
+#####
+def iterations(func):
+    return click.option(
+        "-n",
+        "--iterations",
+        default=10,
+        show_default=True,
+        type=int,
+        help="Number of times to execute each entity ID provided.",
+    )(func)
```

Comment:
If you want, you can also add [`image_tag`](https://github.com/spotify/klio/blob/master/cli/src/klio_cli/options.py#L154-L160) ([klio-exec definition](https://github.com/spotify/klio/blob/master/exec/src/klio_exec/options.py#L72)). It would need to be define / used similarly to `per_element`:

```py
# in core/src/klio_core/options.py
def image_tag(*args, **kwargs):
    def wrapper(func):
        return click.option( 
            "--image-tag",
            help="Docker image tag to use"
            **kwargs
        )(func)

    if args:
        return wrapper(args[0])
    return wrapper

# in exec/src/klio_exec/cli.py
# for all the commands that use the image_tag option:
@core_options.image_tag
...

# in cli/src/klio_cli/cli.py
# for all the commands that use the image_tag option:
@core_options.image_tag(default=None, show_default="``git-sha[dirty?]``")
...
```